equinox-shipment/Domain/FinalizationTransactionProcessor.fs

```diff
@@ -0,0 +1,59 @@
+module FinalizationTransactionProcessor
+
+    type Service(transactions: FinalizationTransaction.Service, containers: Container.Service, shipments: Shipment.Service) =
+
+        // Transaction ID can be a md5 hash of all shipments ID
+        let execute (transactionId: string) : FinalizationTransaction.Events.Event option -> Async<bool> =
+            let rec loop (update: FinalizationTransaction.Events.Event option) =
+                async {
+                    let loop event = loop (Some event)
+
+                    let! action =
+                        transactions.Apply(transactionId, update)
+
+                    match action with
+                    | FinalizationTransaction.Action.AssignShipments (containerId, shipmentIds) ->
+                        let! result =
+                            shipmentIds
```

Comment:
consider `seq { for x in shipmentIds -> async {. ... } }`

Reply:
What's the difference between using `seq { for .. in .. -> async { ... } }` and `Array.map`?

Reply:
Array definitely allocates an array and materialized things - but you werent asked to do that - in the same way you shouldnt `List.map` just because you started from a `list` if the end goal is a `seq`